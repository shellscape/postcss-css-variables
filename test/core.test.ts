import { fixture, process } from './helpers';

const tests = {
  'accept whitespace in var() declarations': 'whitespace-in-var-declaration',
  'replace in scoped blocks': 'background-url',
  'work when no variable name passed to `var()`': 'empty-var-func',
  'work with `!important` variable declarations': 'important-variable-declaration',
  'work with any combinator selector if the last piece is the variable we have in the map':
    'scope-last-piece-of-combinator-sequence',
  'work with locally scoped variable in a non-root rule': 'local-variable-non-root',
  'work with star selector': 'star-selector-scope',
  'work with variables declared in root': 'root-variable',
  'work with variables with parenthesis in fallback': 'fallback-with-parenthesis'
};

describe('core', async () => {
  test.each(Object.entries(tests))('%s', async (_desc, file) => {
    const css = await fixture(`core/${file}`);
    const result = await process(css);
    expect(result).toMatchSnapshot();
  });

  test('throw on malformed var()', async () => {
    const css = await fixture('core/malformed-variable-usage');
    expect(async () => process(css)).rejects.toThrowErrorMatchingSnapshot();
  });
});
