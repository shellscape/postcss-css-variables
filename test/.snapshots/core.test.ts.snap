// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`core > accept whitespace in var() declarations 1`] = `".bar{width:1px;height:1px;margin-top:1px}"`;

exports[`core > replace in scoped blocks 1`] = `".xl\\\\:bg-green-500{background-color:rgb(34 197 94/1)}"`;

exports[`core > throw on malformed var() 1`] = `[CssSyntaxError: <css input>:7:12: Unclosed bracket]`;

exports[`core > work when no variable name passed to \`var()\` 1`] = `".box-empty{width:var()}"`;

exports[`core > work with \`!important\` variable declarations 1`] = `".box-foo{width:210px}.box-bar{width:320px}"`;

exports[`core > work with any combinator selector if the last piece is the variable we have in the map 1`] = `".foo + .bar{width:150px}"`;

exports[`core > work with locally scoped variable in a non-root rule 1`] = `".box-foo{width:150px;.box-bar{width:150px}}"`;

exports[`core > work with star selector 1`] = `".foo{background:#0f0}"`;

exports[`core > work with variables declared in root 1`] = `".box-foo{width:15vw}.box-bar{width:150px}"`;

exports[`core > work with variables with parenthesis in fallback 1`] = `".box{box-shadow:0px 2px 2px 0px #fff}"`;
